### PR TITLE
[ci][microcheck] add macos + windows tests to microcheck

### PR DESCRIPTION
**Pull request by can-anyscale**
Add macos + window tests to microcheck

Test:
- CI

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]
do people ask for this? need this?

like if it is microcheck, probably very unlikely for someone to click the unblock button?

**Comment by can-anyscale** [comment]
i cut a task for @zcin to debug some flaky tests on windows ;); she didn't ask for this but thinking about doing this before she even asks

also https://github.com/ray-project/ray/issues/45278 is another release blocker windows test still under debugging

core team debugs macos tests on PR a lot as well

**Pull request merged**
